factory/apiResolverFactory.go

```diff
@@ -301,8 +302,14 @@ func createScQueryElement(
 	}
 
 	if args.processComponents.ShardCoordinator().SelfId() == core.MetachainShardId {
+		blockChainHookImpl, errBlockChainHook := hooks.NewBlockChainHookImpl(argsHook)
+		if errBlockChainHook != nil {
+			return nil, errBlockChainHook
+		}
+
 		argsNewVmFactory := metachain.ArgsNewVMContainerFactory{
-			ArgBlockChainHook:   argsHook,
+			BlockChainHook:      blockChainHookImpl,
```

Comment:
`CreateApiResolver` uses `AccountsAdapter` and not `AccountsAdapterAPI`. expected?

Reply:
good catch